txcache/selectionSessionWrapper.go

```diff
@@ -0,0 +1,105 @@
+package txcache
+
+import (
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/data"
+)
+
+// After moving "mx-chain-storage-go/txcache" into "mx-chain-go", maybe merge this component into "SelectionSession".
+type selectionSessionWrapper struct {
+	session          SelectionSession
+	recordsByAddress map[string]*accountRecord
+}
+
+type accountRecord struct {
+	initialNonce    uint64
+	initialBalance  *big.Int
+	consumedBalance *big.Int
+}
+
+func newSelectionSessionWrapper(session SelectionSession) *selectionSessionWrapper {
+	return &selectionSessionWrapper{
+		session:          session,
+		recordsByAddress: make(map[string]*accountRecord),
+	}
+}
+
+func (sessionWrapper *selectionSessionWrapper) getAccountRecord(address []byte) *accountRecord {
```

Comment:
is the getAccountRecord called only in the selection context?
can it be called in concurrently?

Reply:
It's not called concurrently - only within selection :+1: